datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -605,6 +605,11 @@ impl LogicalPlan {
         expr.transform(&|expr| {
             match &expr {
                 Expr::Placeholder { id, data_type } => {
+                    if id.is_empty() {
+                        return Err(DataFusionError::Internal(
```

Comment:
What do you think about making this a `DataFusion::Plan` error rather than `Internal` -- it seems to me this isn't a bug in datafusion (which is what `Internal` represents) but rather a bug in the code that called DataFusion.